lib/BuiltinsEnum.h

```diff
@@ -0,0 +1,338 @@
+// Copyright 2019 The Clspv Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CLSPV_LIB_BUILTINSENUM_H_
+#define CLSPV_LIB_BUILTINSENUM_H_
+
+namespace clspv {
+namespace Builtins {
+
+enum BuiltinType {
+  kBuiltinNone = 0,
+
+  kType_Async_Start = 1,
```

Comment:
Why assign values here? It seems just to create a maintenance headache if any builtins are ever added. If you want them sortable based on ranges, I suggest:
```
kBuiltinNone = 0,
kType_Async_Start,
kAsyncWorkGroupCopy = kType_Async_Start,
kWaitGroupEvents,
kPrefetch,
kType_Async_End = kPrefetch,
...
```

Reply:
I agree it can lead to problems but this proposal also can be error prone if adding new enums. I prefer not to refer to other enums. Probably best to rely on incremental enum values and remove all numbers and assignments. Does that sound reasonable?

Reply:
That's ok with me.

Reply:
I think this is still outstanding to change.

Reply:
hmm, thought I updated that last week.. sending shortly

Reply:
ok, updated for real this time.